core/src/main/java/io/grpc/Metadata.java

```diff
@@ -642,9 +645,9 @@ private static String validateName(String n) {
       return n;
     }
 
-    private Key(String name) {
+    private Key(String name, boolean pseudo) {
```

Comment:
Do we want to be extra safe and disallow calling .put() on any keys that are pseudo?
Also, there are users today who try to read ":status", so we should probably pre-define a Key<Integer> for reading ":status".